manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -94,16 +101,16 @@ func (tr *TaskReaper) Run() {
 }
 
 func (tr *TaskReaper) tick() {
-	if len(tr.dirty) == 0 {
+	if len(tr.dirty) == 0 && len(tr.orphaned) == 0 {
 		return
 	}
 
 	defer func() {
 		tr.dirty = make(map[instanceTuple]struct{})
+		tr.orphaned = nil
 	}()
 
-	var deleteTasks []string
-
+	deleteTasks := tr.orphaned
```

Comment:
> I thought we wanted to keep taskreaper as the only entity who should be deleting tasks. 

We also delete tasks in the global and replicated orchestrators if the service they are associated with is deleted.